nexus/test-utils/src/lib.rs

```diff
@@ -254,8 +255,10 @@ impl RackInitRequestBuilder {
         let zone = self
             .internal_dns_config
             .host_zone(
-                // TODO-cleanup use TypedUuid everywhere
-                OmicronZoneUuid::from_untyped_uuid(dataset_id),
+                // XXX why is zone id being created from dataset id?
```

Comment:
This doesn't seem right; maybe related to https://github.com/oxidecomputer/omicron/pull/7000#discussion_r1830280079 (cc @smklein)?

Reply:
12da0ba8e